File: python/test/test_cours.py
from src.cours import Cours

class TestCours:
    def test_creation(self):
        c = Cours("Mathématiques", "M. Dupont")
        assert c.nom == "Mathématiques"
        assert c.professeur == "M. Dupont"

    def test_str(self):
        c = Cours("Mathématiques", "M. Dupont")
        assert "Mathématiques" in str(c)
        assert "M. Dupont" in str(c)

File: java/test/TestCours.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestCours {

    @Test
    public void testCreation() {
        Cours c = new Cours("Mathématiques", "M. Dupont");
        assertEquals("Mathématiques", c.getNom());
        assertEquals("M. Dupont", c.getProfesseur());
    }

    @Test
    public void testStr() {
        Cours c = new Cours("Mathématiques", "M. Dupont");
        assertTrue(c.toString().contains("Mathématiques"));
        assertTrue(c.toString().contains("M. Dupont"));
    }
}
